### PR TITLE
ENH adding alias "regression" and "classification" 

### LINKED ISSUE #1177: Allow `tabular_learner("regression")`
### Problem Description

We can use `skrub.tabular_learner(estimator="regressor")` to use a default `HistGradientBoostingRegressor` as the final estimator of the pipeline.

I find myself typing `skrub.tabular_learner("regression")` more often than I would like :)
I think there is no ambiguity in this case and as a user rather than an error I would prefer skrub to accept "regression" as a synonym for "regressor" -- same for "classification"

### Feature Description

"regression" and "classification" would be accepted to mean "regressor" and "classifier" respectively

### PR DESCRIPTION
**Pull request by mrastgoo**
closes #1177 

Adding "regression" as an alias to "regressor" and "classification" to "classifier"

### CODE REVIEW

File: skrub/_tabular_learner.py [hunk 1]
```diff
@@ -106,24 +106,24 @@ def tabular_learner(estimator, *, n_jobs=None):
 
     We can easily get a default pipeline for regression or classification:
 
-    >>> tabular_learner('regressor')                                    # doctest: +SKIP
+    >>> tabular_learner('regression')                                    # doctest: +SKIP
```
Comment (glemaitre) [inline skrub/_tabular_learner.py:109]: For the later changes here and under, I don't think that we need to change the documentation.

**Review by glemaitre — COMMENTED** [review]
I think that we should add an entry in the `CHANGES.rst` to acknowledge the new options.

File: skrub/_tabular_learner.py [hunk 2]
```diff
@@ -61,14 +61,14 @@ def tabular_learner(estimator, *, n_jobs=None):
```
Comment (glemaitre) [inline skrub/_tabular_learner.py:61]: I think that we need to amend as well the documentation in the long summary just above where we already mentioned `"regressor"` and `"classifier"`.

File: CHANGES.rst [hunk 3]
```diff
@@ -12,6 +12,11 @@ Ongoing development
 Skrub is a very recent package.
 It is currently undergoing fast development and backward compatibility is not ensured.
 
+Major changes
+-------------
+* :func:`tabular_learner` accepts the alias ``"regression"`` for the option ``"regressor"`` and ``"classification"`` for ``"classifier"``.
```
Comment (glemaitre) [inline CHANGES.rst:17]: This should solve the failure about the changelog by mentioning the PR number.

```suggestion
* :func:`tabular_learner` accepts the alias ``"regression"`` for the option
   ``"regressor"`` and ``"classification"`` for ``"classifier"``.
   :pr:`1180` by :user:`Mojdeh Rastgoo <mrastgoo>`.
```

**Review by glemaitre — APPROVED** [review]
LGTM. Thanks @mrastgoo

**Review by Vincent-Maladiere — APPROVED** [review]
LGTM, thanks @mrastgoo :)

**Comment by jeromedockes** [comment]
awesome!! thanks a lot @mrastgoo  :)

**Pull request merged**